src/components/PostsList/PostsList.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import './PostsList.scss';
+import { getAllPosts, getPosts } from '../../api/posts';
+import { Post } from '../../types';
+
+type Props = {
+  userSelectedId: string;
+  selectPost: (arg?: Post) => void;
+  post: Post | undefined;
+};
+
+export const PostsList: React.FC<Props> = ({
+                                             userSelectedId,
+                                             selectPost,
+                                             post,
+                                           }) => {
+  const [currentPostList, setPostList] = useState<Post[]>([]);
+
+  const allPosts = async () => {
+    const result = await getAllPosts();
+
+    setPostList(result);
+  };
```

Comment:
Here we have a similar issue from the `App.tsx` file, what if getAllUsers throws an error? We should be able to handle it and treat it somehow, reducing the impact of the error on the user experience. What if getAllUsers's promise take more than 10 seconds to respond? we should be able to show its loading state to the user.

Also, the handling of async functions are destardalized, on `App.tsx` we use `.then` and `.catch` to handle the response, and here we use `async` and `await` to handle its response, we have to choose one pattern and follow it along the project, but isn't a good practice to use both at the same time.